compiler/syntax/src/res_parens.ml

```diff
@@ -148,12 +148,12 @@ let binary_expr_operand ~is_lhs expr =
       else Nothing)
 
 let sub_binary_expr_operand parent_operator child_operator =
-  let prec_parent = ParsetreeViewer.operator_precedence parent_operator in
-  let prec_child = ParsetreeViewer.operator_precedence child_operator in
+  let open ParsetreeViewer in
+  let prec_parent = operator_precedence parent_operator in
+  let prec_child = operator_precedence child_operator in
   prec_parent > prec_child
-  || prec_parent == prec_child
-     && not
-          (ParsetreeViewer.flattenable_operators parent_operator child_operator)
+  || is_equality_operator parent_operator
```

Comment:
Can you explain this change? Why is precedence not required anymore?

Reply:
https://github.com/rescript-lang/rescript-compiler/blob/7a58f3b7e433e278921ef130c0c006106a20f97c/compiler/syntax/src/res_parsetree_viewer.ml#L357-L364

It was like
`a == b && !(a == b && not (is_eq_a && is_eq_b))`
`a == b && (a !== b || is_eq_a && is_eq_b)`
`a == b && (false || is_eq_a && is_eq_b)`
`a == b && is_eq_a && is_eq_b`

and `a == b` was duplicated so